sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -99,3 +97,67 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double) extends Aggregator[IN, Double, Double] {
+  override def zero: Double = Double.PositiveInfinity
```

Comment:
A modified version of existing `TypedSumDouble` to use `java.lang.Double` as `BUF` and `OUT` looks like below. It returns `null` when applying on an empty dataset.

    class TypedSumDouble[IN](val f: IN => Double)
          extends Aggregator[IN, java.lang.Double, java.lang.Double] {
      override def zero: java.lang.Double = null
      override def reduce(b: java.lang.Double, a: IN): java.lang.Double = {
        if (b == null) {
          f(a)
        } else {
          b + f(a)
        }
      }
      override def merge(b1: java.lang.Double, b2: java.lang.Double): java.lang.Double = {
        if (b1 == null) {
          b2
        } else if (b2 == null) {
          b1
        } else {
          b1 + b2
        }
      }
      override def finish(reduction: java.lang.Double): java.lang.Double = {
        reduction
      }
      override def bufferEncoder: Encoder[java.lang.Double] = ExpressionEncoder[java.lang.Double]()
      override def outputEncoder: Encoder[java.lang.Double] = ExpressionEncoder[java.lang.Double]()

      // Java api support
      def this(f: MapFunction[IN, java.lang.Double]) = this(x => f.call(x))

      def toColumnJava: TypedColumn[IN, java.lang.Double] = {
        toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
      }
    }